src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.cs

```diff
@@ -187,6 +187,30 @@ protected internal override void OpenPin(int pinNumber)
                     throw ExceptionHelper.GetIOException(ExceptionResource.OpenPinError, Marshal.GetLastWin32Error());
                 }
 
+                int mode = Interop.libgpiod.gpiod_line_direction(pinHandle);
+                if (mode == 1)
+                {
+                    pinHandle.PinMode = PinMode.Input;
+                }
+                else if (mode == 2)
+                {
+                    pinHandle.PinMode = PinMode.Output;
+                }
```

Comment:
should you throw on different value?

Reply:
The value of PinMode defaults to input. This shouldn't ever give something else, but I don't like throwing here because if the library does something unexpected, this would make the whole driver unusable.